### PR TITLE
Add .NET 8.0 TFM and use new AesGcm constructor

### PR DESCRIPTION
**Pull request by zzyzy**
Closes https://github.com/dvsekhvalnov/jose-jwt/issues/248

### CODE REVIEW

**Comment by dvsekhvalnov** [comment]
@zzyzy it's failing bunch test targets.

File: jose-jwt/JWTSettings.cs [hunk 1]
```diff
@@ -13,7 +13,7 @@ public class JwtSettings
     {
         public JwtSettings()
         {
-#if NET472 || NETSTANDARD2_1 
+#if NET472 || NETSTANDARD2_1 || NET
```
Comment (dvsekhvalnov) [inline jose-jwt/JWTSettings.cs:16]: shouldn't it be `NET8_0` ? What `NET` specifically targets?

Comment (zzyzy) [reply jose-jwt/JWTSettings.cs:16]: `NET` targets .NET 5.0 and above

**Comment by zzyzy** [comment]
> @zzyzy it's failing bunch test targets.

Sorry for the late response, I've fixed them.

File: jose-jwt/crypto/AesGcmNetCore.cs [hunk 2]
```diff
@@ -37,7 +45,14 @@ public static byte[][] Encrypt(byte[] key, byte[] iv, byte[] aad, byte[] plainTe
         /// <exception cref="CryptographicException">if decryption failed by any reason</exception>
         public static byte[] Decrypt(byte[] key, byte[] iv, byte[] aad, byte[] cipherText, byte[] authTag)
         {
+#if NET
```
Comment (dvsekhvalnov) [inline jose-jwt/crypto/AesGcmNetCore.cs:48]: i guess we can just collapse both branches together now? No need to use .NET 8.0 specific AesGcm() constructor as you check auth tag length explicitly now. And you can leverage `Ensure.BitSize(..)` helper right at the start of the method.

And after just clean up all `NET` targeting, cause no need to and PR will become nice & small, just of couple files :)

File: jose-jwt/JWTSettings.cs [hunk 3]
```diff
@@ -58,7 +58,7 @@ public JwtSettings()
             { JwsAlgorithm.ES256, new EcdsaUsingSha(256) },
             { JwsAlgorithm.ES384, new EcdsaUsingSha(384) },
             { JwsAlgorithm.ES512, new EcdsaUsingSha(521) }
-#elif NETSTANDARD || NET461 || NET472
+#elif NETSTANDARD || NET461 || NET472 || NET
```
Comment (dvsekhvalnov) [inline jose-jwt/JWTSettings.cs:61]: as mentioned in `AesGcmNetCore.cs` comment, no need to touch all project with with extra targeting `|| NET` as fix is just checking length and should work just fine on every plarform.

**Comment by dvsekhvalnov** [comment]
Closed in favor of https://github.com/dvsekhvalnov/jose-jwt/pull/255

**Pull request closed without merging**